src/server_gameloop.rs

```diff
@@ -161,16 +163,24 @@ pub fn gameloop() {
             for enemyid in enemies {
                 let mut enemy = gd.enemies[enemyid].lock().unwrap();
                 let moveloc = enemy.enemy_type.move_pattern();
-                let confirmedloc = enemy.mov(&gd, (Etype::Enemy, enemyid), moveloc, now);
+                assert!(!(*enemy.mut_mov_next() > now));
```

Comment:
why tho

Reply:
IDK it was mostly to make sure things were sane after I changed it. I can remove if you want

Reply:
i would remove, shouldn't be necessary outside of sanity checks while you were implementing the featuers

AND THINK ABOUT THE PERFORMANCE DOWNSIDES